src/app/login-auth.service.ts

```diff
@@ -6,8 +6,13 @@ import { Injectable } from "@angular/core";
 export class LoginAuthService {
   constructor() {}
 
-  onSubmit(loginForm: any) {
-    let password: string = loginForm.value.password;
-    alert(`Username: ${loginForm.value.username} \nPassword: ${password}`);
+  login(username: String, password: String) {
+    return new Promise(function(resolve, reject) {
+      if (!(password === "12345" && username === "hagay")) {
```

Comment:
keep logic clear. 
why to do ! () on the check. 
 if (password === "12345" && username === "hagay") {
       // its ok
else 
     // is not ok

Reply:
yes, i didnt understood that too. VScode did it for me.. so i tought this is the best practice, i will fix it.